### ISSUE TITLE
Spike: research Datadog integration with Next.js

### ISSUE THREAD

**Issue by timcosgrove** [department-of-veterans-affairs/va.gov-cms#16491]
## Requirements

Investigate tools and methods for integrating Datadog and Next.js/Next Build.

```[tasklist]
### Acceptance criteria
- [x] Learn and document how to send logs from Next.js to specified locations rather than STDOUT
- [x] Learn and document how to forward logs from our application to Datadog
- [x] Learn and document if there are integrations with Datadog and Javascript applications or specifically Next.js for sending API calls to Datadog
- [x] Learn and document how to integrate Datadog APM with Next.js
```

## Background & implementation details

Documenting findings as comments in this ticket is sufficient.

Datadog docs: https://docs.datadoghq.com/
Next.js docs: https://nextjs.org/docs

Most likely web search will be key here. 

```[tasklist]
### Implementation tasks
- [ ] Use this if it helps you or feel free to delete.
```

**Comment by jschmidt-civicactions** [comment]
Note to self: RUM (Real User Monitoring) is not needed for this ticket.

**Comment by jschmidt-civicactions** [comment]
VA Confluence pages on Datadog setup for a new app: 
https://vfs.atlassian.net/wiki/spaces/OT/pages/2236612646/Monitor+Applications+and+Infrastructure+with+Datadog

We will need to [create a service account key](https://vfs.atlassian.net/wiki/spaces/OT/pages/1934655507/Connect+an+Application+to+Datadog) for NextBuild and store this in the AWS parameter store.

EKS should already have a [Datadog agent](https://docs.datadoghq.com/containers/kubernetes/distributions/?tab=operator#EKS) available for a NextBuild deployment, but see below for possible additional configuration requirements. 

Kubernetes log collection must be [enabled in the Datadog Helm chart](https://docs.datadoghq.com/containers/kubernetes/log/?tab=helm) (this may already be set by AWS EKS maintainers). The app and api keys should be configured here.

Once enabled in Kubernetes:
- We can configure [custom log collection](https://docs.datadoghq.com/logs/log_collection/nodejs/?tab=winston30) from Node.js to forward specified logs using the Winston logging library.
- We can also configure [metric collection](https://docs.datadoghq.com/integrations/node/#metric-collection) from Node.js. This requires DogStatsD to be enabled for external metrics on the Agent in Kubernetes (will need to verify this config with the EKS maintainers).

Winston 3.x example:
```
const { createLogger, format, transports } = require('winston');

const logger = createLogger({
  level: 'info',
  exitOnError: false,
  format: format.json(),
  transports: [
    new transports.File({ filename: `${appRoot}/logs/<FILE_NAME>.log` }),
  ],
});

module.exports = logger;

// Example logs
logger.log('info', 'Hello simple log!');
logger.info('Hello log with metas',{color: 'blue' });
```

Metrics collection example using dd-trace:
```
// Require dd-trace
const tracer = require('dd-trace').init();

// Increment a counter
tracer.dogstatsd.increment('page.views');
```

**Comment by jschmidt-civicactions** [comment]
Last piece of this that I have not addressed is Datadog APM integration, which I will wrap up this week.

I believe my other questions surrounding configuration are best addressed in #9886. The docs linked here should be a good starting point.

**Comment by jschmidt-civicactions** [comment]
APM data in Datadog:
Looks like the node.js dd-trace npm package will get us what we need see 
https://github.com/DataDog/dd-trace-js
and
https://hix.dev/tutorials/react-next-js/monitoring-with-datadog
